src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -49,9 +48,15 @@ trait RegularSync {
       waitingForActor = None
       handleBlockBodies(peer, blockBodies)
 
-    case block: BroadcastBlocks if broadcasting =>
+    case BroadcastBlocks(newBlocks) if broadcasting =>
       //FIXME: Decide block propagation algorithm (for now we send block to every peer) [EC-87]
-      peersToDownloadFrom.keys.foreach(_.ref ! block)
+      val blocksToSendToEachPeer = for {
+        peer <- peersToDownloadFrom.keys
```

Comment:
I like your idea :+1:
Maybe it's better to create a separate task for this or add it in PR #191 as various changes were already done to `PeerEventBus` in that PR?